src/code_80091750.c

```diff
@@ -1079,361 +1078,214 @@ extern s8 gTextColor;
+            else if ((characterWidthIndex != -2) && (characterWidthIndex == -1)) {
+                stringWidth += ((tracking + 7) * x_scale);
             }
-            goto loop_2;
-        }
-        // Duplicate return node #33. Try simplifying control flow for better match
-        return;
-    }
-block_11:
-    if (arg6 != 1) {
-        if (arg6 != 2) {
-            if (arg6 != 3) {
-                if (arg6 != 4) {
-
-                } else {
-                    goto block_17;
-                }
+            else{
+                return;
+            }
+            if (characterWidthIndex >= 0x30) {
+                text += 2;
             } else {
-                goto block_16;
+                text += 1;
             }
-        } else {
-block_17:
-            phi_s3 = arg0 - (phi_s1 / 2);
-        }
-    } else {
-block_16:
-        phi_s3 = arg0 - phi_s1;
+        } while(*text != 0);
     }
-    if (arg6 < 3) {
-        sp60 = 1;
-    } else {
-        sp60 = 2;
+
+    if ((arg6 == 1) || (arg6 == 3)){
+        column -= stringWidth;
     }
-    temp_v0_2 = gDisplayListHead;
-    gDisplayListHead = temp_v0_2 + 8;
-    temp_v0_2->unk0 = 0x6000000;
-    temp_v0_2->unk4 = &D_020077A8;
-    if (*arg2 != 0) {
-loop_22:
-        temp_v0_3 = get_ascii_char_width_index(phi_s2);
-        if (temp_v0_3 >= 0) {
-            temp_s0 = (temp_v0_3 * 4) + &D_800E7E84;
-            func_80099184(segmented_to_virtual(*temp_s0));
-            gDisplayListHead = func_8009BEF0(gDisplayListHead, segmented_to_virtual(*temp_s0), phi_s3, arg1, sp60, arg4, arg5);
-            phi_s3 += (*(&gCharacterWidthMap + (temp_v0_3 * 2)) + arg3) * arg4;
-            goto block_28;
-        }
-        if ((temp_v0_3 != -2) && (temp_v0_3 == -1)) {
-            phi_s3 += (arg3 + 7) * arg4;
-block_28:
-            if (temp_v0_3 >= 0x30) {
-                phi_s2_2 = phi_s2 + 2;
-            } else {
-                phi_s2_2 = phi_s2 + 1;
+    else if ((arg6 == 2) || (arg6 == 4)){
+        column -= (stringWidth / 2);
+    }
+
+    gDisplayListHead += 1;
+    gDisplayListHead->words.w0 = 0x6000000;
+    gDisplayListHead->words.w1 = &D_020077A8;
+    s32 temp;
+    if (*text != 0) {
+        do{
+            characterWidthIndex = get_ascii_char_width_index(text);
+            if (characterWidthIndex >= 0) {
+                temp = segmented_to_virtual(D_800E7E84[characterWidthIndex]);
+                func_80099184(temp);
+                gDisplayListHead = func_8009BEF0(gDisplayListHead, temp, column, row, arg6 < 3 ? 1 : 2, x_scale, y_scale);
+                column += (gCharacterWidthMap[characterWidthIndex] + tracking) * x_scale;
             }
-            phi_s2 = phi_s2_2;
-            if (*phi_s2_2 == 0) {
-                goto block_32;
+            else if ((characterWidthIndex != -2) && (characterWidthIndex == -1)) {
+                column += (tracking + 7) * x_scale;
             }
-            goto loop_22;
-        }
-        temp_v0_4 = gDisplayListHead;
```

Comment:
Unless this scenario is somehow different...
Whenever you see this pattern, it's a gfx macro that sends graphics commands straight to the gdp.

Take unk0 put it through `./gfxdis.f3dex` (found on github/n64-master) along with the argument. They must be 32 bits each: ex:
`06000000020077D8`

```
./gfxdis.f3dex -d 06000000020077D8
{
    gsSPDisplayList(0x020077D8),
}
```
Now we have to make this work in code:
Step 1: gsSP -> gSP (Do this for all commands when you're working in code or .text instead of .data. This is called dynamic display lists verses static display lists like what you see in the trophy file)
Next gDisplayListHead is always added as an argument with ++
    gSPDisplayList(gDisplayListHead++, 0x020077D8);

Sometimes there's an actual function hidden between a bunch of gfx. It's easy to spot, but if you're not looking for it it's easily missed.

Reply:
I'll give it a shot, the display list stuff is the major contributor to the non-matchiness of those functions so I'm all for fixing it. Thank you for the info!

Reply:
Something that may be useful for future reference
You can make gfxdis.f3dex output in the desired format automatically by using the -g option
```
gfxdis.f3dex -g gDisplayListHead++ -d 06000000020077A8 06000000020077D8
gSPDisplayList(gDisplayListHead++, 0x020077A8);
gSPDisplayList(gDisplayListHead++, 0x020077D8);
```

Reply:
Waaaaaat?!? Seriously? *facepalm* I've been doing it manually this whole time.

Also,  since I missed this step: If `0x020077A8` is a variable/symbol then you can put `D_020077A8`. It's just you can't put the D_ part into gfxdis. You could even ignore typing in the symbol into gfxdis if you just want to remember to add it as the second argument. You'd still have to put eight zeros though.